clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -146,11 +146,17 @@ private void addMetadataListener() {
         this.metadata.addListener(new Metadata.Listener() {
             @Override
             public void onMetadataUpdate(Cluster cluster) {
-                // if we encounter any unauthorized topics, raise an exception to the user
-                if (!cluster.unauthorizedTopics().isEmpty())
-                    throw new TopicAuthorizationException(new HashSet<>(cluster.unauthorizedTopics()));
 
                 if (subscriptions.hasPatternSubscription()) {
+
+                    Set<String> unauthorizedTopics = new HashSet<String>();
+                    for (String topic : cluster.unauthorizedTopics()) {
+                        if (subscriptions.getSubscribedPattern().matcher(topic).matches())
```

Comment:
This condition should probably be like a few lines below:

```java
                         if (subscriptions.getSubscribedPattern().matcher(topic).matches() &&
                                 !(excludeInternalTopics && TopicConstants.INTERNAL_TOPICS.contains(topic)))
```

So we would want to extract that condition to a helper method perhaps.